rust/src/mqtt/detect.rs

```diff
@@ -353,7 +353,7 @@ pub extern "C" fn rs_mqtt_tx_get_publish_message(
 
 #[no_mangle]
 pub extern "C" fn rs_mqtt_tx_get_subscribe_topic(tx: &MQTTTransaction,
-    i: u16,
+    i: u32,
```

Comment:
Is the message id 32 or 16 bits?

This states 16-bits but perhaps I'm misreading?  https://public.dhe.ibm.com/software/dw/webservices/ws-mqtt/mqtt-v3r1.html#msg-id

Reply:
You are referring to the internal message ID used to connect multiple items in a transaction. The `i` here, however, is a local ID coming from the C side of the detect code: https://github.com/OISF/suricata/blob/9ef13d309bae377b10e0554b8d37471518af1580/src/detect-mqtt-subscribe-topic.c#L83

There's no direct connection to the message ID, other than that the range of the message ID should set a limit for the the maximum number of items to be traversed. But I wanted to rather be safe than sorry and just use an `int` here.

Reply:
Sorry, did not notice your 👍🏻  here -- does this mean we can resolve this question?

Reply:
For me, it is ok.
@jlucovsky is it ok for you ?